cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go

```diff
@@ -1704,58 +1705,85 @@ func TestPackageAppVersionsSummary(t *testing.T) {
 func TestParsePluginConfig(t *testing.T) {
 	testCases := []struct {
 		name                    string
-		pluginYAMLConf          string
+		pluginYAMLConf          []byte
 		exp_versions_in_summary VersionsInSummary
+		exp_error               error
 	}{
 		{
-			name:                    "empty plugin config",
-			pluginYAMLConf:          ``,
+			name:                    "non existing plugin-config file",
+			pluginYAMLConf:          nil,
 			exp_versions_in_summary: VersionsInSummary{0, 0, 0},
+			exp_error:               ErrFileNotFound,
```

Comment:
...which means you won't be able to just check for a canned error that you've created, but rather you need to check for the actual error returned by the underlying call. But that's fine, because the errors package gives you [errors.Is](https://pkg.go.dev/github.com/pkg/errors#Is), which keeps unwrapping an error and checking. So in this case, the exp_error will be whatever the underlying `ioutil.Readfile` returns, which you'll need to find out. Could be an `fs.ErrNotExist` or `fs.PathError`, not sure.

Reply:
After chatting on slack, I took a look and although the above (using `errors.Is`) works fine for the missing file, it doesn't for the `json.Unmarshal` error. We can find the error easily enough by grepping for "cannot unmarshal":

```
➜  v1alpha1 git:(versionInSummary_3722) ✗ rg -i "cannot unmarshal" /usr/local/go/src/encoding/json
/usr/local/go/src/encoding/json/decode_test.go
2182:   {new(net.IP), "json: cannot unmarshal number into Go value of type *net.IP"},

/usr/local/go/src/encoding/json/decode.go
134:            return "json: cannot unmarshal " + e.Value + " into Go struct field " + e.Struct + "." + e.Field + " of type " + e.Type.String()
136:    return "json: cannot unmarshal " + e.Value + " into Go value of type " + e.Type.String()
150:    return "json: cannot unmarshal object key " + strconv.Quote(e.Key) + " into unexported field " + e.Field.Name + " of type " + e.Type.String()
```

Checking `decode.go` [shows that it's an `UnmarshalTypeError`](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/encoding/json/decode.go;l=134), which you'd think would solve the problem, but it doesn't, because the `UnmarshalTypeError` has a pointer receiver, which means the use of `errors.Is` will effectively be doing: `errors.Is(err, &json.UnmarshalTypeError{})`, which, looking at the implementation of `errors.Is`, repeatedly unwraps the `err`, checking each time for equality with the target. But because the target is a pointer, [equality means that they point to the same address](https://golang.org/ref/spec#Comparison_operators), so `errors.Is` always returns false. It appears that `errors.Is` is only intended to work with non-pointer-receivers.

So yes, rather than working around that, as much as we dislike comparing error strings, probably a good balance of pragmatism and effort to just compare the error string for the test :)

Reply:
Thanks for understanding. As discussed on slack, I shall use string.Contains() for comparing and update the PR.